app/src/main/java/com/example/xpvehicles/activities/MainActivity.java

```diff
@@ -118,17 +106,17 @@ private void setUserLocation() {
             public void onLocationChanged(@NonNull Location location) {
                 Double userLongitude = location.getLongitude();
                 Double userLatitude = location.getLatitude();
-                userLocation = userLatitude + " " + String.valueOf(userLongitude);
-                Log.i(TAG, "User's location is " + userLocation);
-                explore_fragment.notifyAdapter();
+                userLocation = userLatitude + " " + userLongitude;
+                Log.i(TAG, "User's location is " + userLocation); exploreFragment.notifyAdapter();
             }
         });
     }
 
+    public static boolean isValidEmail(CharSequence target) {
+        return (!TextUtils.isEmpty(target) && Patterns.EMAIL_ADDRESS.matcher(target).matches());
+    }
```

Comment:
You shouldn't make MainActivity to have static method like this. If it is used by many classes, let's create a class called "EmailChecker" or something. If it is used by this class only, let's just make it private regular method only